src/core/thread/mle.hpp

```diff
@@ -323,6 +324,17 @@ class Mle : public InstanceLocator, private NonCopyable
      */
     bool IsMinimalEndDevice(void) const { return mDeviceMode.IsMinimalEndDevice(); }
 
+    /**
+     * This method gets the Network Data type (full set or stable subset) that this device requests.
+     *
+     * @returns The Network Data type requested by this device.
+     *
+     */
+    NetworkData::Type GetNetworkDataType(void) const
```

Comment:
This method effectively does the same thing as `IsMinimalEndDevice()`. Should we remove the latter and use `GetNetworkDataType()` everywhere?

Reply:
Good idea. 
We may need to add some more changes (e.g. add a helper method to get version value for a `Type`, i.e., `GetVersion(Type aType)`). I think it may make the overall code simpler (easier to read). I will try changing this later.

Reply:
Changed this in the new push.

Reply:
Looks great, thanks!